
syntax = "proto3";

package shed_avail_service;

message ScheduleDetails {
    int32 booking_id = 1;
    string current_status = 2;
    int32 mechanic_id= 3;
    int32 schedule_id = 4;
}

message ScheduleResponse {
    string status  = 1;
}

message AvailabilityRequirements {
    int32 required_hours = 1;
}

message Slot {
    string available_date = 1;
    repeated int32 available_hours= 2;
}

message AvailabilityResponse {
    repeated Slot slots = 1;
}

message MechStatus {
    int32 mech_id = 1;
    string status = 2;
}

message empty{

}

message Status {
    int32 status = 1;
}

service ShedAvailService {
    rpc simulate_daily_schedule(empty) returns (Status) {};
    rpc simulate_reminder_emails(empty) returns (Status) {};
    rpc get_available_hours(AvailabilityRequirements) returns (AvailabilityResponse) {};
    rpc handle_schedule(ScheduleDetails) returns (ScheduleResponse) {};
    rpc updated_mech_status(MechStatus) returns (empty) {};

}